lib/modules/manager/pixi/artifacts.ts

```diff
@@ -0,0 +1,100 @@
+import is from '@sindresorhus/is';
+import { TEMPORARY_ERROR } from '../../../constants/error-messages';
+import { logger } from '../../../logger';
+import { exec } from '../../../util/exec';
+import type { ExecOptions } from '../../../util/exec/types';
+import {
+  deleteLocalFile,
+  ensureCacheDir,
+  getSiblingFileName,
+  readLocalFile,
+  writeLocalFile,
+} from '../../../util/fs';
+import { getGitEnvironmentVariables } from '../../../util/git/auth';
+import { Result } from '../../../util/result';
+import type { UpdateArtifact, UpdateArtifactsResult } from '../types';
+import { commandLock, pickConfig } from './lockfile';
+import { LockfileYaml } from './schema';
+
+export async function updateArtifacts({
+  packageFileName,
+  updatedDeps,
+  newPackageFileContent,
+  config,
+}: UpdateArtifact): Promise<UpdateArtifactsResult[] | null> {
+  logger.debug(`pixi.updateArtifacts(${packageFileName})`);
+  const { isLockFileMaintenance } = config;
+
+  if (!is.nonEmptyArray(updatedDeps) && !isLockFileMaintenance) {
+    logger.debug('No updated pixi deps - returning null');
+    return null;
+  }
+
+  const cmd: string[] = [];
+
+  const lockFileName = getSiblingFileName(packageFileName, 'pixi.lock');
+  const existingLockFileContent = await readLocalFile(lockFileName, 'utf8');
+  if (!existingLockFileContent) {
+    logger.debug(`No lock file found`);
+    return null;
+  }
+  logger.debug(`Updating ${lockFileName}`);
+
+  const { val } = Result.parse(existingLockFileContent, LockfileYaml).unwrap();
+  const cfg = pickConfig(val?.version);
+  const constraint = cfg?.range;
```

Comment:
prefer `config.constraints?.pixi` over this. so only read and parse lockfile if no user constrains are there